src/Examples/OnSlideExample.tsx

```diff
@@ -0,0 +1,40 @@
+import type { FC, PropsWithChildren, ComponentType } from 'react';
+import { useRef, useState } from 'react';
+import { Slider, SliderTypes } from '../Slider';
+
+const OnSlideExample: FC<typeof Slider extends ComponentType<infer T> ? PropsWithChildren<T> : never> = (sliderProps) => {
```

Comment:
Nitpicking, but in the current state this still allows `children` to be passed even though they're not rendered

Reply:
Nitpicking part 2: Since it's in the src folder, typescript types will also be generated even though there's not much point in doing so